RIOT-OS/RIOT
# cpu/cc26xx_cc13xx: add & link CCFG configuration

### Contribution description

The CCFG is the Customer Configuration, this enables us to change it at compile time.

- Adds a linker script for `cc26x2_cc13x2` processors, describing the memory regions correctly.
  - The ROM size was changed from `352K` to `360360` bytes (`351.9140625K` or `0x57FA8`) which is the real size.
- Adds CCFG vendor code.
- Adds `FORCE_VDDR_HH` configuration option. Enables the usage of +14 dBm PA at the cost of higher current consumption. Disabled by default.

More options when needed (such as the bootloader) can added.

The `cc26x2_cc13x2` based boards pass the `cortexm_common_ldscript` test, should I whitelist them?

### Testing procedure

- Code should still flash normally without problems.
- `CFLAGS="-DCONFIG_CC26XX_CC13XX_UPDATE_CCFG=1" make flash BOARD=cc1312-launchpad`
- Or any other combination using `make menuconfig` and selecting the option to update the CCFG.

### Issues/PRs references

See also #13635
